src/vs/vscode.d.ts

```diff
@@ -8994,6 +9016,12 @@ declare module 'vscode' {
 		 */
 		export function registerTerminalLinkProvider(provider: TerminalLinkProvider): Disposable;
 
+		/**
+		 * Registers a provider for a contributed terminal profile.
+		 * @param id The ID of the contributed terminal profile.
```

Comment:
[nit] Add a note about the contribution/activation event in the docs

Reply:
related to that: do you enforce that only the extension that contributed the profile (via package.json) is allows to use the id when?

Reply:
No this isn't enforced yet